### PR TITLE
chore(security): add security.txt file

### PR DESCRIPTION
**Pull request by yakkomajuri**
I think we should just do this.

### CODE REVIEW

File: posthog/views.py [hunk 1]
```diff
@@ -76,9 +72,20 @@ def stats(request):
...
+    ROBOTS_TXT_CONTENT = (
+        "User-agent: *\nDisallow: /shared_dashboard/" if settings.MULTI_TENANCY else "User-agent: *\nDisallow: /"
+    )
     return HttpResponse(ROBOTS_TXT_CONTENT, content_type="text/plain")
 
 
+def security_txt(request):
+    SECURITY_TXT_CONTENT = """
+        Contact: engineering@posthog.com
+        Hiring: https://posthog.com/careers
```
Comment (Twixes) [inline posthog/views.py:84]: I thought the `Expires` field is mandatory too (according to https://securitytxt.org/)?

Comment (yakkomajuri) [inline posthog/views.py:84]: 2 year expire buffer? or should we revisit this once a year? @Twixes 

```suggestion
        Contact: mailto:engineering@posthog.com
        Hiring: https://posthog.com/careers
        Expires: 2024-03-14T00:00:00.000Z
```

File: posthog/urls.py [hunk 2]
```diff
@@ -106,6 +106,7 @@ def opt_slash_path(route: str, view: Callable, name: Optional[str] = None) -> UR
     opt_slash_path("batch", capture.get_event),
     opt_slash_path("s", capture.get_event),  # session recordings
     opt_slash_path("robots.txt", robots_txt),
+    opt_slash_path("security.txt", security_txt),
```
Comment (guidoiaquinti) [inline posthog/urls.py:109]: As per proposed [spec](https://securitytxt.org/) I think it should be placed at `.well-known/security.txt`

Comment (yakkomajuri) [inline posthog/urls.py:109]:
```suggestion
    opt_slash_path(".well-known/security.txt", security_txt),
```

**Review by guidoiaquinti — APPROVED** [review]

**Pull request merged**